esx_service/vmdk_ops.py

```diff
@@ -490,24 +492,33 @@ def get_vol_path(datastore, tenant_name=None):
     if os.path.isdir(path):
         # If the path exists then return it as is.
         logging.debug("Found %s, returning", path)
-        return path
+        return path, None
 
     if not os.path.isdir(dock_vol_path):
         # The osfs tools are usable for DOCK_VOLS_DIR on all datastores
         cmd = "{0} {1}".format(OSFS_MKDIR_CMD, dock_vol_path)
         rc, out = RunCommand(cmd)
-        if rc != 0:
-            logging.warning("Failed to create %s", dock_vol_path)
-            return None
+        errMsg = None
+        if rc == ERR_OSFS_MKDIR_READONLY:
```

Comment:
Rather than special casing error codes, why not use existing library function to translate error code to string? I am not a python expert but tried following and seems to be fine;

>>> os.strerror(13)
'Permission denied'
>>> os.strerror(30)
'Read-only file system'